src/components/upload-queue/QueueItem.tsx

```diff
@@ -0,0 +1,52 @@
+import { Component, createMemo } from 'solid-js'
```

Comment:
I would just put all of the components in the same file if they aren't going to be used elsewhere in the app